overlord/snapstate/reboot_test.go

```diff
@@ -605,6 +612,37 @@ func (s *rebootSuite) TestArrangeSnapTaskSetsLinkageAndRestartBootBaseAndOtherBa
 	c.Check(s.taskSetIsTransactional(tss[2]), Equals, false)
 }
 
+func (s *rebootSuite) TestArrangeSnapTaskSetsLinkageForSnapWithBaseAndWithout(c *C) {
+	defer snapstatetest.MockDeviceModel(MakeModel20("brand-gadget", nil))()
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	tss := []*state.TaskSet{
+		s.taskSetForSnapSetup("snap-base", "", snap.TypeBase),
```

Comment:
I expected also to look into the case where one of the bases is also the boot base

Reply:
I was unsure that we needed to test that, added a unit test case for this as well!